account_payment_partner/models/account_move.py

```diff
@@ -56,7 +56,8 @@ def _compute_partner_bank_filter_type_domain(self):
     @api.depends("partner_id", "company_id")
     def _compute_payment_mode(self):
         for move in self:
-            move.payment_mode_id = False
+            if move.company_id and move.payment_mode_id.company_id != move.company_id:
+                move.payment_mode_id = False
```

Comment:
This is missing the other part that assigns `move.payment_mode_id = move.payment_mode_id`.

Reply:
In 14.0 is not necessary.

Reply:
Is there a test that proves it?

Reply:
The test `test_account_move_payment_mode_id_default` that is added in this PR proves it.
However, the problem in 13.0 is that in a compute it is necessary to define all the options and always assign a value (or a CacheMiss error appears); on the other hand, in 14.0, if there is any case in which no value is defined, odoo already defines one without showing an error.

Reply:
Are you talking about a default value or the previous one?

Reply:
Everything is related to default value in computed fields.

Reply:
But then it's needed, because we don't want to switch to the default value, but keep the previous one. There's something that is not fitting me about what you comment... Take into account that this method can be executed multiple times, not only on new record, but it's acting like an onchange.